src/services/BaseService.ts

```diff
@@ -12,7 +15,26 @@ function getType(object: object | null) {
 }
 
 export default abstract class BaseService {
-	// region public static methods
+
+	private static queues: Queue[] = [];
+	private transporters: PlatformTransporter[] = [];
+	private name: string = 'NoName';
+
+	protected constructor(name: string) {
+		this.name = name;
```

Comment:
If using only PushService or MailService as name, this._serviceType() will respond a similar value which is defined in https://github.com/hpi-schul-cloud/node-notification-service/blob/9813ae91659a1ebcec8da593e221c2ef7e08e61e/src/services/BaseService.ts#L142 and overridden in https://github.com/hpi-schul-cloud/node-notification-service/blob/a605005c89f5f3134523a893946adca636bf81e4/src/services/MailService.ts#L69 and https://github.com/hpi-schul-cloud/node-notification-service/blob/a605005c89f5f3134523a893946adca636bf81e4/src/services/PushService.ts#L57

Reply:
This is true. I see it later. But not sure it it is the why that we should go. 
In escalation logic i add a other name for mail server like for the base mail service to see which redis connection and query is used, or throw an error.
In future we must outsource the job-query logic from BaseService that we only have one job-query and one redis connection to solve the read SCAN/write errors. If this is done we can remove the name attribute if we want.